rust/syntect-plugin/src/main.rs

```diff
@@ -214,6 +214,9 @@ impl<'a> PluginState<'a> {
             let tab_size = ctx.get_config().tab_size;
             let buf_size = ctx.get_buf_size();
             if let Some(line) = line_num.and_then(|idx| ctx.get_line(idx).ok()) {
+                // do not send update if last line is empty string (contains only new line)
+                if line == "\n" { return None; }
```

Comment:
This is basically it, yea, but there's one subtlety: Instead of comparing directly to "\n", we should compare against the line-ending in use by this particular file; it may be using something else. You can see we do something similar above here, when we check `is_newline`.

You'll have to do something to get around the borrow checker here; perhaps cloning the config's `line_ending`.

And a tiny nit: Rust style is to leave off the trailing ';' on return statements.

otherwise looks good. :)